File: factories/processor_factory.py

from services.processors.transferencia_processor import TransferenciaProcessor
from services.processors.deposito_processor import DepositoProcessor
from services.processors.saque_processor import SaqueProcessor

class ProcessorFactory:
    @staticmethod
    def criar_processor(tipo, conta_repository, logger):
        if tipo == 'TRANSFERENCIA':
            return TransferenciaProcessor(conta_repository, logger)
        elif tipo == 'DEPOSITO':
            return DepositoProcessor(conta_repository, logger)
        elif tipo == 'SAQUE':
            return SaqueProcessor(conta_repository, logger)
        else:
            raise ValueError(f"Tipo de processor não suportado: {tipo}")

File: factories/ProcessorFactory.js

const TransferenciaProcessor = require('../services/processors/TransferenciaProcessor');
const DepositoProcessor = require('../services/processors/DepositoProcessor');
const SaqueProcessor = require('../services/processors/SaqueProcessor');

class ProcessorFactory {
    static criarProcessor(tipo, contaRepository, logger) {
        switch (tipo.toUpperCase()) {
            case 'TRANSFERENCIA':
                return new TransferenciaProcessor(contaRepository, logger);
            case 'DEPOSITO':
                return new DepositoProcessor(contaRepository, logger);
            case 'SAQUE':
                return new SaqueProcessor(contaRepository, logger);
            default:
                throw new Error(`Tipo de processor '${tipo}' não suportado`);
        }
    }
}

module.exports = ProcessorFactory;
